core/lib/zksync_core/src/consistency_checker/mod.rs

```diff
@@ -358,9 +413,50 @@ impl ConsistencyChecker {
             .await?)
     }
 
+    async fn sanity_check_validator_timelock_addr(&self) -> Result<(), CheckError> {
+        let Some(address) = self.validator_timelock_addr else {
+            return Ok(());
+        };
+        tracing::debug!("Performing sanity checks for validator timelock contract {address:?}");
+
+        let call =
+            CallFunctionArgs::new("getName", ()).for_contract(address, self.contract.clone());
```

Comment:
Not sure what exactly should we call in the contract; I went with the simplest option with human-readable output. 

For some reason, it looks like not all `view` functions from the contract declaration at `self.contract` (i.e., [`IZkSync`](https://github.com/matter-labs/era-contracts/blob/main/l1-contracts/contracts/zksync/interfaces/IZkSync.sol)) are actually present in the validator timelock contract; is this expected? E.g., I've initially wanted to fetch the protocol version from it (part of the `IGetters` interface), but it's not actually present.

Reply:
Timelock validator only implements `IExecutor`. If you switch to using DiamondProxy then all getters from IGetters will be achievable